# Backport of https://github.com/bazel-contrib/rules_foreign_cc/pull/1570 (commit 4462c3ed23fb),
# "toolchains: load cc_import in the glib/gettext spoke BUILD files".
#
# Bazel 9 removed the native cc_import, so these generated BUILD files fail
# analysis without an explicit load(). Upstream moved this code to
# toolchains/private/source_spokes.bzl; on 0.15.1 it still lives in
# toolchains/built_toolchains.bzl, so the same change is applied there.
--- a/toolchains/built_toolchains.bzl
+++ b/toolchains/built_toolchains.bzl
@@ -268,7 +268,7 @@
     maybe(
         http_archive,
         name = "glib_dev",
-        build_file_content = '''
+        build_file_content = '''load("@rules_cc//cc:defs.bzl", "cc_import")
 cc_import(
     name = "glib_dev",
     hdrs = glob(["include/**"]),
@@ -286,7 +286,7 @@
     maybe(
         http_archive,
         name = "glib_src",
-        build_file_content = '''
+        build_file_content = '''load("@rules_cc//cc:defs.bzl", "cc_import")
 cc_import(
     name = "msvc_hdr",
     hdrs = ["msvc_recommended_pragmas.h"],
@@ -326,7 +326,7 @@
     maybe(
         http_archive,
         name = "gettext_runtime",
-        build_file_content = '''
+        build_file_content = '''load("@rules_cc//cc:defs.bzl", "cc_import")
 cc_import(
     name = "gettext_runtime",
     shared_library = "bin/libintl-8.dll",
